src/client/WebSocketClient.ts

```diff
@@ -364,11 +364,19 @@ export class WebSocketClient extends EventEmitter {
   private pingTime: number;
   private readonly pongTime: number;
 
+  private _subscriptions: {
+    [channel_name: string]: Required<WebSocketChannel>;
```

Comment:
I don't know of a way to limit an index signature object to just the values of an enum (`WebSocketChannelName`). Maybe you know how to wrangle TypeScript to do that?

I used index signature as my master store because I believe its must more performant than continually looping through an array to find items.